### PR TITLE
AG-3028 lock down image tags

### PR DESCRIPTION
**Pull request by david-martin**

### CODE REVIEW

**Review by odra — APPROVED** [review]

File: roles/provision-buildfarm-apb/tasks/artifact-proxy-operator.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@
         value: "{{ proxy_route_host.stdout }}"
       - name: JENKINS_HOSTNAME
         value: "{{ jenkins_route_host.stdout }}"
-      image: 'docker.io/aerogear/artifact-proxy-operator:latest'
+      image: 'docker.io/aerogear/artifact-proxy-operator:1.0.0-alpha'
```
Comment (david-martin) [inline roles/provision-buildfarm-apb/tasks/artifact-proxy-operator.yml:57]: @philbrookes can you ping when this tag is available

File: roles/provision-buildfarm-apb/defaults/main.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 android_sdk_home: "{{ lookup('env', 'ANDROID_HOME')|default('/opt/android-sdk-linux', True) }}"
-android_slave_image: "docker.io/aerogear/digger-android-slave-image:latest"
+android_slave_image: "docker.io/aerogear/digger-android-slave-image:1.0.0-alpha"
```
Comment (david-martin) [inline roles/provision-buildfarm-apb/defaults/main.yml:3]: I pulled 'latest' and tagged/pushed this new tag.

File: roles/provision-buildfarm-apb/vars/main.yml [hunk 3]
```diff
@@ -16,6 +16,6 @@ deployments:
   - name: "{{ ag_android_sdk.deployment.name }}"
     containers:
     - name: "android-sdk"
-      image: "docker.io/aerogear/digger-android-sdk-image:latest"
+      image: "docker.io/aerogear/digger-android-sdk-image:1.0.0-alpha"
```
Comment (david-martin) [inline roles/provision-buildfarm-apb/vars/main.yml:19]: I pulled 'latest' and tagged/pushed this new tag.

File: roles/provision-buildfarm-apb/tasks/artifact-proxy-operator.yml (hunk 1 above)
Comment (odra) [reply roles/provision-buildfarm-apb/tasks/artifact-proxy-operator.yml:57]: Just ran `docker pull  docker.io/aerogear/artifact-proxy-operator:1.0.0-alpha` and the image was pulled.

**Comment by psturc** [comment]
I tried to provision it on remote cluster, it failed with the error:
```

TASK [provision-buildfarm-apb : Copy gradlew to remote temp folder] ************
--
  | changed: [localhost]
  | TASK [provision-buildfarm-apb : Copy required tarball] *************************
  | fatal: [localhost]: FAILED! => {"changed": false, "cmd": ["oc", "-n", "cicd", "rsync", "/tmp/aerogear-digger-installer/", "android-sdk-1-vs9kc:/opt/android-sdk-linux/tools/templates/gradle/wrapper/", "--include=gradle-wrapper.tar.gz"], "delta": "0:00:00.946441", "end": "2018-06-08 12:55:02.774773", "msg": "non-zero return code", "rc": 1, "start": "2018-06-08 12:55:01.828332", "stderr": "rsync: chgrp \"/opt/android-sdk-linux/tools/templates/gradle/wrapper/.\" failed: Operation not permitted (1)\nrsync: chgrp \"/opt/android-sdk-linux/tools/templates/gradle/wrapper/.gradle-wrapper.tar.gz.LL2DLT\" failed: Operation not permitted (1)\nrsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1052) [sender=3.0.9]\nerror: exit status 23", "stderr_lines": ["rsync: chgrp \"/opt/android-sdk-linux/tools/templates/gradle/wrapper/.\" failed: Operation not permitted (1)", "rsync: chgrp \"/opt/android-sdk-linux/tools/templates/gradle/wrapper/.gradle-wrapper.tar.gz.LL2DLT\" failed: Operation not permitted (1)", "rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1052) [sender=3.0.9]", "error: exit status 23"], "stdout": "sending incremental file list\n./\ngradle-wrapper.tar.gz\n\nsent 53183 bytes  received 34 bytes  106434.00 bytes/sec\ntotal size is 53073  speedup is 1.00", "stdout_lines": ["sending incremental file list", "./", "gradle-wrapper.tar.gz", "", "sent 53183 bytes  received 34 bytes  106434.00 bytes/sec", "total size is 53073  speedup is 1.00"]}
  | PLAY RECAP *********************************************************************
  | localhost                  : ok=41   changed=17   unreachable=0    failed=1


```

**Comment by odra** [comment]
@psturc  minishift or cluster up?

**Comment by psturc** [comment]
@odra it's actually OCP: https://wendy-ocp-slave2.skunkhenry.com:8443/console/project/localregistry-aerogear-digger-apb-prov-wjbkw/browse/pods/apb-15bf30c6-44eb-4a9b-8743-4489f5e2b987?tab=logs Do you think the error is related to OpenShift version?

I'm also currently testing it locally with `oc cluster up` from https://github.com/aerogear/mobile-core/pull/88

**Comment by odra** [comment]
could be, which version is ocp using? (my local vm is using openshift 3.9)

**Comment by psturc** [comment]
@odra it's 3.7. I'm still testing it locally. It takes eternity to finish.

Stuck on: `TASK [provision-buildfarm-apb : Copy Jenkins plugin install script to host] ****`

Update: ok it's stuck because the OpenShift instance just died quietly a while ago lol

**Comment by odra** [comment]
that step usually takes a while :)

**Review by psturc — APPROVED** [review]
Haven't tested locally, because docker got stuck on my machine during provisioning (probably not enough free space), but images can be pulled, so if @odra's local test finishes successfully I'm ok with merging this.

**Comment by odra** [comment]
Provisioning worked in my local vm

**Pull request merged**